plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -353,47 +355,37 @@ private DeltaLakeTransactionLogEntry buildAddEntry(ConnectorSession session, Blo
         Block addEntryBlock = block.getObject(pagePosition, Block.class);
         log.debug("Block %s has %s fields", block, addEntryBlock.getPositionCount());
 
-        Map<String, String> partitionValues = getMap(addEntryBlock, 1);
-        long size = getLong(addEntryBlock, 2);
-        long modificationTime = getLong(addEntryBlock, 3);
-        boolean dataChange = getByte(addEntryBlock, 4) != 0;
-        Map<String, String> tags = getMap(addEntryBlock, 7);
-
-        String path = getString(addEntryBlock, 0);
-        AddFileEntry result;
-        if (!addEntryBlock.isNull(6)) {
-            result = new AddFileEntry(
-                    path,
-                    partitionValues,
-                    size,
-                    modificationTime,
-                    dataChange,
-                    Optional.empty(),
-                    Optional.of(parseStatisticsFromParquet(addEntryBlock.getObject(6, Block.class))),
-                    tags);
-        }
-        else if (!addEntryBlock.isNull(5)) {
-            result = new AddFileEntry(
-                    path,
-                    partitionValues,
-                    size,
-                    modificationTime,
-                    dataChange,
-                    Optional.of(getString(addEntryBlock, 5)),
-                    Optional.empty(),
-                    tags);
-        }
-        else {
-            result = new AddFileEntry(
-                    path,
-                    partitionValues,
-                    size,
-                    modificationTime,
-                    dataChange,
-                    Optional.empty(),
-                    Optional.empty(),
-                    tags);
-        }
+        int totalFieldCount = addEntryBlock.getPositionCount();
+        int fieldId = 0;
+        String path = getString(addEntryBlock, fieldId++);
+        Map<String, String> partitionValues = getMap(addEntryBlock, fieldId++);
+        long size = getLong(addEntryBlock, fieldId++);
+        long modificationTime = getLong(addEntryBlock, fieldId++);
+        boolean dataChange = getByte(addEntryBlock, fieldId++) != 0;
+        Optional<String> stats = Optional.empty();
+        if (!addEntryBlock.isNull(fieldId)) {
```

Comment:
This part of the change has the side effect that if both json and parquet stats are available we will parse and store both, while we used to prioritize Parquet stats and skip parsing the json ones if not needed.

I'm worried about the memory implications because of that. In the case where there are many entries, this may increase the memory requirements by quite a lot.